example/directives/authorization/authorization.go

```diff
@@ -31,11 +31,15 @@ type HasRoleDirective struct {
 	Role string
 }
 
+func (h *HasRoleDirective) AllowLocation(l string) bool {
+	return l == "FIELD_DEFINITION"
+}
+
 func (h *HasRoleDirective) ImplementsDirective() string {
 	return "hasRole"
 }
 
-func (h *HasRoleDirective) Validate(ctx context.Context, _ interface{}) error {
+func (h *HasRoleDirective) Validate(ctx context.Context, _, _ interface{}) error {
```

Comment:
Just thinking out loud - should this method be called `Validate`? Can these do something than validation? I know that as per the PR title and your intentions is to add validation, however, I'm trying just to think if this can be more general. Still don't have a good use case, though 🤔